rj_gameplay/rj_gameplay/play/penalty_offense.py

```diff
@@ -1,75 +1,61 @@
-    Dict,
-    List,
-    Tuple,
-    Type,
-)
-import numpy as np
+from rj_gameplay.tactic import prep_move
 
 
-class PenaltyOffense(play.IPlay):
-    """Move all robots to our half, but away from ball to prep for penalty kick"""
+class State(Enum):
+    INIT = auto()
+    PREP = auto()
+    READY = auto()
+    # DONE = auto()
 
+
+class PenaltyOffense(stp.play.Play):
     def __init__(self):
-        self.tactics = [
-            goalie_tactic.GoalieTactic(True),
-            striker_tactic.LineKickStrikerTactic(None),
-            move_tactic.Move((1.8, 0.0), priority=role.Priority.LOW),
-            move_tactic.Move((2.1, 0.0), priority=role.Priority.LOW),
-            move_tactic.Move((2.4, 0.0), priority=role.Priority.LOW),
-            move_tactic.Move((2.7, 0.0), priority=role.Priority.LOW),
-        ]
-        self.role_assigner = NaiveRoleAssignment()
+        super().__init__()
 
-    def compute_props(self, prev_props):
-        pass
+        self._state = State.INIT
 
     def tick(
         self,
         world_state: rc.WorldState,
-        prev_results: role.assignment.FlatRoleResults,
-        props,
-    ) -> Tuple[
-        Dict[Type[tactic.SkillEntry], List[role.RoleRequest]],
-        List[tactic.SkillEntry],
-    ]:
-        # Get role requests from all tactics and put them into a dictionary
-        role_requests: play.RoleRequests = {
-            tactic: tactic.get_requests(world_state, None) for tactic in self.tactics
-        }
-        # Flatten requests and use role assigner on them
-        flat_requests = play.flatten_requests(role_requests)
-        flat_results = self.role_assigner.assign_roles(
-            flat_requests, world_state, prev_results
-        )
-        role_results = play.unflatten_results(flat_results)
-        # Get list of all SkillEntries from all tactics
-        skills = []
-        tac_index = 0
-        for tactic in self.tactics:
-            if tac_index == 1:  # TODO BAD HACK
-                skills += tactic.tick(world_state, role_results[tactic])
-            else:
-                skills += tactic.tick(world_state, role_results[tactic])
-            tac_index += 1
-        # Get all role assignments
-        # SkillEntry to (list of?) RoleResult
-        skill_dict = {}
-        for tactic in self.tactics:
-            skill_dict.update(role_results[tactic])
-        return (skill_dict, skills)
+    ) -> List[RobotIntent]:
+
+        if self._state == State.INIT:
+            self.prioritized_tactics = [
+                prep_move.PrepMove(world_state)
+                # assume line tactic is working
+                # line_tactic.LineTactic(world_state),
+                # line_tactic.LineTactic(world_state),
+                # line_tactic.LineTactic(world_state),
+                # line_tactic.LineTactic(world_state),
+                # line_tactic.LineTactic(world_state),
+            ]
+            self.assign_roles(world_state)
+            self._state = State.PREP
```

Comment:
you should add a while loop here that checks if the number of requested tactics is actually assigned before switching the state, see this line:

https://github.com/RoboJackets/robocup-software/pull/1855/files#diff-a0f92563cb784149768f1724a5d7e152b2ce3b8042d3477f11ce84f823ddb589R111